File: DomainLayer/Models/Single/RatingModel.cs
using System.ComponentModel.DataAnnotations;

namespace GroceryWala.DomainLayer.Models.Single
{
    public  class RatingModel
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "user id is required")]
        public string UserId { get; set; }

        [Required(ErrorMessage = "Product Id required")]
        public string ProductId { get; set; }

        [Required(ErrorMessage = "Rating of product is required")]
        public int Rating { get; set; }
    }

}

File: GroceryWala_UI/src/app/Models/RatingModel.ts
export class RatingModel{
    id?:string;
    userId:string;
    productId:string;
    rating:number = 0;
    constructor(userId:string, productId:string, rating:number, id?:string) {
        this.userId = userId;
        this.productId = productId;
        this.rating = rating;
        this.id = id;
    }
}